internal/config/config.go

```diff
@@ -46,17 +49,18 @@ type Fetcher struct {
 }
 
 type Config struct {
-	Benchmark             string                  `config:"config.v1.benchmark"`
-	Type                  string                  `config:"config.v1.type"`
-	Deployment            string                  `config:"config.v1.deployment"`
-	CloudConfig           CloudConfig             `config:"config.v1"`
-	KubeConfig            string                  `config:"kube_config"`
-	Period                time.Duration           `config:"period"`
-	Processors            processors.PluginConfig `config:"processors"`
-	BundlePath            string                  `config:"bundle_path"`
-	PackagePolicyId       string                  `config:"package_policy_id"`
-	PackagePolicyRevision int                     `config:"revision"`
-	Index                 string                  `config:"index"`
+	Benchmark              string                  `config:"config.v1.benchmark"`
+	Type                   string                  `config:"config.v1.type"`
+	Deployment             string                  `config:"config.v1.deployment"`
+	AssetInventoryProvider string                  `config:"config.v1.asset_inventory_provider"`
```

Comment:
I understand why you did this way.

From a first thought, I had in mind to re-use the `benchmark` field, that is used to define the "provider" on CSPM.

I know they are different concepts. But they somehow feel connected to me. It's a "sub selection" of the type.

Do you think there is a way to make a generic config here?

As I type I realize that to update `benchmark` config key could lead to few changes on integrations and maybe even kibana. So I'm not sure if it's a good idea. But I'm still posting so you give some thought too :)

Reply:
Yeah, I was refraining from `Benchmark` for integration's sake. Another way I was considering was using the `Type`, as you noticed. It's currently `asset_inventory`, but we could make it `asset_inventory_<provider>` and apply logic to that. WDYT?

Reply:
i could see a separate issue where we use something like `Kind` to express both `Benchmark` and `AssetInventoryProvider` (as in - a flavor's sub-type). that'll require some `.hbs` file changes, cloudbeat back-compatbility config resolver, and if we really want to clean up stuff - some non-mandatory minor kibana changes 

although it might be too much trouble just for tidiness sake